output/types.go

```diff
@@ -75,10 +75,17 @@ type WithTestRunStop interface {
 	SetTestRunStopCallback(func(error))
 }
 
-// WithRunStatusUpdates means the output can receive test run status updates.
-type WithRunStatusUpdates interface {
+// WithStopWithTestError allows output to receive the error value that the test
+// finished with. It could be nil, if the test finished nominally.
+//
+// If this interface is implemented by the output, StopWithError() will be
+// called instead of Stop().
+//
+// TODO: refactor the main interface to use this method instead of Stop()? Or
+// something else along the lines of https://github.com/grafana/k6/issues/2430 ?
+type WithStopWithTestError interface {
 	Output
-	SetRunStatus(latestStatus lib.RunStatus)
+	StopWithTestError(testRunErr error) error // nil testRunErr means error-free test run
```

Comment:
```suggestion
	StopWithTestError(ctx context.Context, testRunErr error) error // nil testRunErr means error-free test run
```

I would use this opportunity to introduce a graceful stop. I'm ok with just changing the API here and addressing the graceful stop from the output manager in a dedicated PR after the removal.

Reply:
hmm, the convention so far has been to always wait for outputs to finish stopping :thinking: that is, unless a user hits Ctrl+C twice to trigger the hard stop with `os.Exit()`, k6 will always wait for outputs to finish... which makes some sense to me and I am not sure we should change it, since we then will need some way to configure this timeout

so, I can add a context here, but we probably won't ever use it... similar to what I said in https://github.com/grafana/k6/pull/2810#discussion_r1083779161, my vote would be to add a note in https://github.com/grafana/k6/issues/2430 and we can discuss it when we completely refactor the outputs :thinking:

Reply:
>hmm, the convention so far has been to always wait for outputs to finish stopping

It doesn't sound to be aligned with the comment here: https://github.com/grafana/k6/issues/2430#issuecomment-1073803076

>This will allow us to set a soft limit for how long outputs are allowed to stop.

I'm ok with moving this as a discussion for 2430